build.sbt

```diff
@@ -10,3 +21,48 @@ val bsp = project
       "org.scalameta" %% "lsp4s" % "0.2.0"
     )
   )
+
+lazy val bsp4j = project
+  .in(file("bsp4j"))
+  .settings(
+    crossVersion := CrossVersion.disabled,
+    autoScalaLibrary := false,
+    javacOptions.in(Compile) ++= List(
+      "-Xlint:all",
+      "-Werror"
+    ),
+    javacOptions.in(Compile, doc) := List("-Xdoclint:none"),
+    javaHome.in(Compile) := inferJavaHome(),
+    javaHome.in(Compile, doc) := inferJavaHome(),
+    TaskKey[Unit]("codegen") := {
+      val compiler = XtendInjectorSingleton.INJECTOR.getInstance(classOf[XtendBatchCompiler])
+      val classpath = dependencyClasspath.in(Compile).value.map(_.data).mkString(File.pathSeparator)
+      compiler.setClassPath(classpath)
+      val sourceDir = sourceDirectory.in(Compile).value / "java"
+      compiler.setSourcePath(sourceDir.getCanonicalPath)
+      val outDir = sourceDirectory.in(Compile).value / "xtend-gen"
+      IO.delete(outDir)
+      compiler.setOutputPath(outDir.getCanonicalPath)
+      object XtendError
+          extends Exception(s"Compilation of Xtend files in $sourceDir failed.")
+          with sbt.internal.util.FeedbackProvidedException
+      if (!compiler.compile())
+        throw XtendError
+    },
+    unmanagedSourceDirectories.in(Compile) += sourceDirectory.in(Compile).value / "xtend-gen",
+    resourceGenerators.in(Test) += Def.task {
```

Comment:
Resource generators are run on every `compile` invocation, even if you only compile a file. I don't think it's necessary for this PR but it would be better if we only codegen when the xtend sources have changed.

Reply:
The xtend code generation needs to be triggered manually via `sbt bsp4j/codegen` whenever an `*.xtend` file is changed. This is a resource generator for the tests, it only writes the working directory of the bloop test directory, which I expect is cheap.